src/expr/src/scalar/func.rs

```diff
@@ -2125,6 +2125,72 @@ fn parse_ident<'a>(
     })?)
 }
 
+fn string_to_array<'a>(
+    string_datum: Datum<'a>,
+    delimiter: Datum<'a>,
+    null_string: Datum<'a>,
+    temp_storage: &'a RowArena,
+) -> Result<Datum<'a>, EvalError> {
+    let string = string_datum.unwrap_str();
+
+    if delimiter.is_null() {
+        let split_all_chars_delimiter = "";
+        return string_to_array_impl(string, split_all_chars_delimiter, null_string, temp_storage);
+    }
+
+    let delimiter = delimiter.unwrap_str();
+    if delimiter.is_empty() {
+        let mut row = Row::default();
+        let mut packer = row.packer();
+        packer.try_push_array(
+            &[ArrayDimension {
+                lower_bound: 1,
+                length: 1,
+            }],
+            vec![string].into_iter().map(Datum::String),
+        )?;
+
+        Ok(temp_storage.push_unary_row(row))
+    } else {
+        string_to_array_impl(string, delimiter, null_string, temp_storage)
+    }
+}
+
+fn string_to_array_impl<'a>(
+    string: &str,
+    delimiter: &str,
+    null_string: Datum<'a>,
+    temp_storage: &'a RowArena,
+) -> Result<Datum<'a>, EvalError> {
+    let mut row = Row::default();
+    let mut packer = row.packer();
+    let empty_flags = "";
+    let regex = build_regex(delimiter, empty_flags)?;
+
+    let found = mz_regexp::regexp_split_to_array(string, &regex);
```

Comment:
The regex here works, but it might be faster and easier to just use [`std::str::split`](https://doc.rust-lang.org/std/primitive.str.html#method.split) any particular reason you used a regex?

Reply:
Note that `build_regex` is expensive (see https://github.com/MaterializeInc/database-issues/issues/6199#issuecomment-2764228653 for a measurement), so `std::str::split` would be preferable (if its semantics match).

Reply:
gotcha, will reimplement into std::str::split

Reply:
done